test/unit-tests/expression/function/evaluate.test.js

```diff
@@ -84,4 +84,101 @@ describe('evaluate', function () {
+      assert.strictEqual(math.evaluate('5 ?? null ?? 42'), 5)
+      assert.strictEqual(math.evaluate('null ?? null ?? null ?? 99'), 99)
+    })
+
+    it('should handle nullish coalescing with correct precedence', function () {
+      // ?? has lower precedence than arithmetic operators
+      assert.strictEqual(math.evaluate('null ?? 1 + 2'), 3) // null ?? (1 + 2)
+      assert.strictEqual(math.evaluate('null ?? 2 * 3'), 6) // null ?? (2 * 3)
+      assert.strictEqual(math.evaluate('null ?? 2^3'), 8) // null ?? (2^3)
+
+      // ?? has same precedence as logical OR, left associative
+      assert.strictEqual(math.evaluate('null ?? false or true'), true) // (null ?? false) or true
+      assert.strictEqual(math.evaluate('true or null ?? 42'), true) // (true or null) ?? 42 = true ?? 42 = true
+
+      // Parentheses can override precedence
+      assert.strictEqual(math.evaluate('null ?? (1 + 2)'), 3)
+      assert.strictEqual(math.evaluate('(null ?? 1) + 2'), 3) // (null ?? 1) + 2 = 1 + 2 = 3
+    })
+
+    it('should handle nullish coalescing with complex expressions', function () {
+      const scope = { a: null, b: 5, c: 0 }
+      assert.strictEqual(math.evaluate('a ?? b * 2', scope), 10) // null ?? (5 * 2)
+      assert.strictEqual(math.evaluate('c ?? b * 2', scope), 0) // 0 ?? (5 * 2) = 0
+      assert.strictEqual(math.evaluate('(a ?? b) * 2', scope), 10) // (null ?? 5) * 2 = 10
+    })
+
+    it('should handle nullish coalescing with strings', function () {
+      assert.strictEqual(math.evaluate('null ?? "hello"'), 'hello')
+      assert.strictEqual(math.evaluate('"world" ?? "hello"'), 'world')
+      assert.strictEqual(math.evaluate('"" ?? "hello"'), '') // empty string is not nullish
+    })
+
+    it('should handle nullish coalescing with matrices and arrays', function () {
+      assert.deepStrictEqual(math.evaluate(['null ?? 1', '2 ?? null', 'null ?? null ?? 3']), [1, 2, 3])
+      assert.deepStrictEqual(math.evaluate(math.matrix(['null ?? 1', '2 ?? null'])), math.matrix([1, 2]))
+    })
+
+    it('should handle nullish coalescing with function calls', function () {
+      const scope = {
+        getValue: function () { return null },
+        getDefault: function () { return 42 }
+      }
+      assert.strictEqual(math.evaluate('getValue() ?? getDefault()', scope), 42)
+
+      const scope2 = {
+        getValue: function () { return 10 },
+        getDefault: function () { return 42 }
+      }
+      assert.strictEqual(math.evaluate('getValue() ?? getDefault()', scope2), 10)
+    })
+
+    it('should handle nullish coalescing with conditional expressions', function () {
+      // Conditional has lower precedence than nullish coalescing
+      assert.strictEqual(math.evaluate('true ? null ?? 5 : 10'), 5) // true ? (null ?? 5) : 10
+      assert.strictEqual(math.evaluate('false ? null ?? 5 : 10'), 10)
+      assert.strictEqual(math.evaluate('null ?? true ? 5 : 10'), 5) // (null ?? true) ? 5 : 10
+    })
```

Comment:
This last expression is another example where the result does not depend on the precedence of nullish coalescing and ternary conditional operators. Please add or replace with an example that does.